cli/azd/pkg/apphost/generate.go

```diff
@@ -23,6 +24,7 @@ const RedisContainerAppService = "redis"
 
 const DaprStateStoreComponentType = "state"
 const DaprPubSubComponentType = "pubsub"
+const containerAppSecretRef = "secretRef:"
```

Comment:
The in-band signaling here feels like it might be a little fragile (imagine someone had an env var who's value was a literal like `secretRef:foo`, would the right thing happen end to end)?

I suspect long term (perhaps when we do an end to end secrets push in preview 4) we'll want to change this strategy here in favor of perhaps using go types to segement secrets from non secrets, but this seems great for Preview 3?